qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,95 @@
+    """
+    try:
+        error_context.context("Start miscellaneous HPT test...", logging.info)
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        session = vm.wait_for_login(timeout=600)
+        get_cmd = "cat /sys/kernel/debug/powerpc/hpt_order"
+        set_cmd = "echo %s > /sys/kernel/debug/powerpc/hpt_order"
+        st, get_hpt_def = session.cmd_status_output(get_cmd, timeout=30)
+        if st:
+            test.cancel("Can't find available HPT value so skip test")
+        else:
+            if params.get("mistype") == "mem":
+                mem_sub_hpt(test, vm)
+            hpt_size = int(get_hpt_def)
+            logging.debug("Default hpt size : '%s'" % get_hpt_def)
+            increment = params.get("increment").split()
+            for increm in increment:
+                hpt_size = hpt_size + int(increm)
+                set_hpt(vm, hpt_size, set_cmd)
+                get_hpt(test, session, hpt_size, get_cmd)
+            if "reboot" in params.get("mistype"):
+                vm.reboot()
+                vm.verify_alive()
+                session = vm.wait_for_login(timeout=600)
+                get_hpt(test, session, int(get_hpt_def), get_cmd)
+            vm.verify_kernel_crash()
+    except Exception as detail:
+        test.fail("Failed in HPT's miscellaneous tests"
+                  " Detail: '%s'" % detail)
+
+
+def set_hpt(vm, hpt_size, set_cmd):
+    """Run this step to set HPT in guest"""
+    session = vm.wait_for_login(timeout=600)
+    session.cmd(set_cmd % (hpt_size), timeout=60)
+
+
+def get_hpt(test, session, hpt_size, get_cmd):
+    """Run this step to check HPT in guest"""
+    get_hpt_value = session.cmd_output(get_cmd, timeout=60)
+    if int(get_hpt_value) != hpt_size:
+        test.fail("HPT expected results aren't consistent")
+
+
+def mem_sub_hpt(test, vm):
+    """Run this step to hotplug 1G memory to guest"""
+    error_context.context("hotplug 1G memory before resizing HPT")
+    mem_id = utils_misc.generate_random_string(8)
+    dimm_id = utils_misc.generate_random_string(8)
+    o = vm.monitor.cmd_qmp("object-add", {
+                           "qom-type": "memory-backend-ram",
+                           "id": "%s" % mem_id,
+                           'props': {'policy': 'default', 'size': 1073741824}})
+    logging.debug("Memory_backend : '%s'" % o)
+    o = vm.monitor.cmd_qmp("device_add", {
+                           "driver": "pc-dimm",
+                           "id": "%s" % dimm_id,
+                           "memdev": "%s" % mem_id})
+    logging.debug("Pc-dimm : '%s'" % o)
+    output = str(vm.monitor.info("memory-devices"))
+    logging.debug("Output of hotplugged memory : '%s'" % output)
+    if not re.search(r"%s" % mem_id, output, re.I):
+        test.fail("hotplugged memory backend can't be found ")
+    if not re.search(r"%s" % dimm_id, output, re.I):
+        test.fail("hotplugged memory backend can't be found ")
```

Comment:
It's better to make line 93 and 95 different, so when if one of them fails, people will know it is  related to `mem_id` or `dimm_id`